### ISSUE TITLE
Forbid propTypes by name

### ISSUE THREAD

**Issue by thany** [yannickcr/eslint-plugin-react#2663]
I would like to be able to specify certain *names* of propTypes (not keys in `PropTypes`) that may not have a propType. A use-case for this is the `children` prop. While the `react/prop-types` can be set u to require propType definitions for all props used, one to skip is the `children` prop. This is an internal/magical prop, so it is already guaranteed to be of the right prop. Setting its propType might actually be slightly confusing.

So with `react/prop-types` configured, this would require a propType definition for `children`:
```jsx
const AwesomeComponent = ({ children, ...rest }) => {
  // Awesome component
};
```
Which has caused folks in my team to go like this:
```jsx
AwesomeComponent.propTypes = {
  children: PropTypes.node,
};
AwesomeComponent.defaultProps = {
  children: null,
};
```
Which is not useful. Leaving this propType and accompaning out will yield an error.

**Comment by ljharb** [comment]
How is that not useful? children can be any value unless you limit it to, for example, node.

**Comment by thany** [comment]
No, children is set to a correct value internally by React.

Unless of course, you'd make a component that really does have an explicit children prop, but that would be... unwise, because then you'd have no way of accessing the React children.

**Comment by ljharb** [comment]
What I mean is, you can pass a function as `children`, for example, or a Symbol, or *literally anything*. It's critically important to have a propType for `children` on custom components, even if it's just `PropTypes.node`.

The only thing React does is on *html elements*, where it either rejects the value or accepts it.

**Comment by thany** [comment]
In that case, a rule to force `children` to be `PropTypes.node`, and its defaultProp to be `undefined`.

**Comment by ljharb** [comment]
So your use case is specifically blocking function children?

**Comment by thany** [comment]
I suppose so, which is not specifically limited to prop-types. It seems like a bit of a broader rule.

**Comment by ljharb** [comment]
The problem with that is that you'd want to block it in the component definition, but not in consumers, since you might be interfacing with a component that you don't control that requires function children, like `React.createContext`.

**Comment by thany** [comment]
Right, that makes sense. I suppose I'll find a better solution them.
